grayskull/strategy/parse_poetry_version.py

```diff
@@ -236,3 +236,109 @@ def encode_poetry_version(poetry_specifier: str) -> str:
+
+
+def encode_poetry_platform_to_selector_item(poetry_platform: str) -> str:
+    """
+    Encodes Poetry Platform specifier as a Conda selector.
+
+    Example: "darwin" => "osx"
+    """
+
+    platform_selectors = {"windows": "win", "linux": "linux", "darwin": "osx"}
+    poetry_platform = poetry_platform.lower().strip()
+    if poetry_platform in platform_selectors:
+        return platform_selectors[poetry_platform]
+    else:  # unknown
+        return ""
+
+
+def encode_poetry_python_version_to_selector_item(poetry_specifier: str) -> str:
+    """
+    Encodes Poetry Python version specifier as a Conda selector.
+
+    Example: ">=3.8,<3.12" => "py>=38 or py<312"
+
+    # handle exact version specifiers correctly
+    >>> encode_poetry_python_version_to_selector_item("3.8")
+    "py==38"
+    >>> encode_poetry_python_version_to_selector_item("==3.8")
+    "py==38"
+    >>> encode_poetry_python_version_to_selector_item("!=3.8")
+    "py!=38"
+
+    # handle caret operator correctly
+    >>> encode_poetry_python_version_to_selector_item("^3.10")
+    # renders '>=3.10.0,<4.0.0'
+    "py>=310 or py<4"
+
+    # handle tilde operator correctly
+    >>> encode_poetry_python_version_to_selector_item("~3.10")
+    # renders '>=3.10.0,<3.11.0'
+    "py>=310 or py<311"
+    """
+
+    if not poetry_specifier:
+        return ""
+
+    version_specifier = encode_poetry_version(poetry_specifier)
+
+    conda_clauses = version_specifier.split(",")
+
+    conda_selectors = []
+    for conda_clause in conda_clauses:
+        operator, version = parse_python_version(conda_clause)
+        version_selector = version.replace(".", "")
+        conda_selectors.append(f"py{operator}{version_selector}")
+    selectors = " or ".join(conda_selectors)
+    return selectors
+
+
+def parse_python_version(selector: str):
+    """
+    Return operator and normalized version from a version selector
+
+    Examples:
+        ">=3.8"   -> ">=", "3.8"
+        ">=3.8.0" -> ">=", "3.8"
+        "<4.0.0"  -> "<", "4"
+        "3.12"    -> "==", 3.12"
+        "=3.8"    -> "==", "3.8"
+
+    The version is normalized to "major.minor" (drop patch if present)
+    or "major" if minor is 0
+    """
+    # Regex to split operator and version
+    pattern = r"^(?P<operator>\^|~|>=|<=|!=|==|>|<|=)?(?P<version>\d+(\.\d+){0,2})$"
+    match = re.match(pattern, selector)
+    if not match:
+        raise ValueError(f"Invalid version selector: {selector}")
+
+    # Extract operator and version
+    operator = match.group("operator")
+    # Default to "==" if no operator is provided or "="
+    operator = "==" if operator in {None, "="} else operator
+    version = match.group("version")
+
+    # Split into major, minor, and discard the rest (patch or additional parts)
+    major, minor, *_ = version.split(".")
+
+    # Return only major if minor is "0", otherwise return major.minor
+    return operator, major if minor == "0" else f"{major}.{minor}"
```

Comment:
I think we probably need this to also support:
```
parse_python_version('>=3')
```
When I tried that, I see (as I expected I might):
```
  File "/home/xylar/code/grayskull/poetry-deps-w-python-ver-platform-and-multiple-deps/./test.py", line 6, in <module>
    print(parse_python_version('>=3'))
          ~~~~~~~~~~~~~~~~~~~~^^^^^^^
  File "/home/xylar/code/grayskull/poetry-deps-w-python-ver-platform-and-multiple-deps/grayskull/strategy/parse_poetry_version.py", line 324, in parse_python_version
    major, minor, *_ = version.split(".")
    ^^^^^^^^^^^^^^^^
ValueError: not enough values to unpack (expected at least 2, got 1)
```
Could you add a test for this situation and make the necessary changes to support a single digit as well?

Reply:
Added support for 1 digit and three dots case (e.g. `3.8.0.1`, never seen for Python version specifier but valid syntax). Thanks for checking that out!